src/index.test.tsx

```diff
@@ -160,6 +164,224 @@ describe("<InputField>", () => {
 
     expect(screen.getByText("hello")).toBeInTheDocument();
   });
+
+  it('should render "component" prop in scope w/ date type', () => {
+    const atom = fieldAtom<Date | null>({ name: "date", value: null });
+    render(
+      <InputField
+        atom={atom}
+        type="date"
+        component={(props) => <input data-testid="input" {...props} />}
+      />
+    );
+    expect(screen.getByTestId("input")).toHaveAttribute("type", "date");
+  });
+
+  it('should render "component" prop in scope w/ datetime-local type', () => {
+    const atom = fieldAtom<Date | null>({ value: null });
+    render(
+      <InputField
+        atom={atom}
+        type="datetime-local"
+        component={(props) => <input data-testid="input" {...props} />}
+      />
+    );
+    expect(screen.getByTestId("input")).toHaveAttribute(
+      "type",
+      "datetime-local"
+    );
+  });
+
+  it('should render "component" prop in scope w/ month type', () => {
+    const atom = fieldAtom<Date | null>({ value: null });
+    render(
+      <InputField
+        atom={atom}
+        type="month"
+        component={(props) => <input data-testid="input" {...props} />}
+      />
+    );
+    expect(screen.getByTestId("input")).toHaveAttribute("type", "month");
+  });
+
+  it('should render "component" prop in scope w/ week type', () => {
+    const atom = fieldAtom<Date | null>({ value: null });
+    render(
+      <InputField
+        atom={atom}
+        type="week"
+        component={(props) => <input data-testid="input" {...props} />}
+      />
+    );
+    expect(screen.getByTestId("input")).toHaveAttribute("type", "week");
+  });
+
+  it('should render "component" prop in scope w/ time type', () => {
+    const atom = fieldAtom<Date | null>({ value: null });
+    render(
+      <InputField
+        atom={atom}
+        type="time"
+        component={(props) => <input data-testid="input" {...props} />}
+      />
+    );
+    expect(screen.getByTestId("input")).toHaveAttribute("type", "time");
+  });
+
+  it('should render "component" prop in scope w/ file type', () => {
+    const atom = fieldAtom<FileList | null>({ value: null });
+    render(
+      <InputField
+        atom={atom}
+        type="file"
+        component={(props) => <input data-testid="input" {...props} />}
+      />
+    );
+    expect(screen.getByTestId("input")).toBeInTheDocument();
+  });
+
+  it('should render "component" prop in scope w/ number type', () => {
+    const atom = fieldAtom({ value: 0 });
```

Comment:
This will render input with value `0` while usually we want to display empty/pristine inputs to the users.
"" as empty value works usually, but is easily validated by zod, so I use `undefined`.